### PR TITLE
Reset stuck low detection

### PR DESCRIPTION
**Pull request by CanyonCasa**
ref: issue #7859

Added a second read to onewireio reset function to verify reset line transitions to high state to detect stuck low faults. A return value of True indicates no devices present (i.e. no presence pulse detected) or a stuck low or stuck high fault.

### CODE REVIEW

File: shared-module/onewireio/OneWire.c [hunk 1]
```diff
@@ -59,8 +59,10 @@ bool common_hal_onewireio_onewire_reset(onewireio_onewire_obj_t *self) {
     common_hal_mcu_delay_us(70);
     bool value = common_hal_digitalio_digitalinout_get_value(&self->pin);
     common_hal_mcu_delay_us(410);
+    // test if bus returned high (idle) and not stck at low
```
Comment (tannewt) [inline shared-module/onewireio/OneWire.c:62]:
```suggestion
    // test if bus returned high (idle) and not stuck at low
```

**Review by tannewt — CHANGES_REQUESTED** [review]
One typo. Looks good otherwise. Thanks!

File: shared-module/onewireio/OneWire.c (hunk 1 above)
Comment (microdev1) [reply shared-module/onewireio/OneWire.c]: I am surprised `codespell` didn't catch that.

**Review by tannewt — APPROVED** [review]
Thank you!

**Pull request merged**